geonode/base/models.py

```diff
@@ -1644,45 +1608,232 @@ def maintenance_frequency_title(self):
+    # types allowed as input for Contact role properties
+    CONTACT_ROLE_USER_PROFILES_ALLOWED_TYPES = Union[settings.AUTH_USER_MODEL, QuerySet, List[settings.AUTH_USER_MODEL]]
+
+    def __set_contact_role_element__(self, user_profile: CONTACT_ROLE_USER_PROFILES_ALLOWED_TYPES, role: str):
+        """_summary_: general setter for all contact roles except owner in resource base
+
+        Args:
+            user_profile (CONTACT_ROLE_USER_PROFILES_ALLOWED_TYPES): _description_
+            role (str): tring coresponding to ROLE_VALUES in geonode/people/enumarations, defining which propery is to set
         """
-        Set metadata_author and/or point of contact (poc) to a resource when any of them is missing
+
+        def __create_role__(
+            resource, role: str, user_profile: settings.AUTH_USER_MODEL
+        ) -> List[settings.AUTH_USER_MODEL]:
+            return ContactRole.objects.create(role=role, resource=resource, contact=user_profile)
+
+        if isinstance(user_profile, QuerySet):
+            ContactRole.objects.filter(role=role, resource=self).delete()
+            return [__create_role__(self, role, user) for user in user_profile]
+        elif isinstance(user_profile, get_user_model()):
+            ContactRole.objects.filter(role=role, resource=self).delete()
+            return __create_role__(self, role, user_profile)
+        elif isinstance(user_profile, list) and all(isinstance(x, get_user_model()) for x in user_profile):
+            ContactRole.objects.filter(role=role, resource=self).delete()
+            return [__create_role__(self, role, profile) for profile in user_profile]
+        else:
+            logger.error(f"Bad profile format for role: {role} ...")
+
+    def get_defined_multivalue_contact_roles(self) -> List[Tuple[List[settings.AUTH_USER_MODEL], str]]:
+        """_summary_: Returns all set contact roles of the ressource with additional ROLE_VALUES from geonode.people.enumarations.ROLE_VALUES. Mainly used to generate output xml more easy.
+
+        Returns:
+              _type_: List[Tuple[List[people object], roles_label_name]]
+              _description: list tuples including two elements: 1. list of people have a certain role. 2. role label
         """
-        if not self.metadata_author:
-            self.metadata_author = self.owner
-        if not self.poc:
-            self.poc = self.owner
+        return {
+            role.label: self.__getattribute__(role.name)
+            for role in Roles.get_multivalue_ones()
+            if self.__getattribute__(role.name)
+        }
+
+    def __get_poc__(self) -> List[settings.AUTH_USER_MODEL]:
+        return self.__get_contact_role_elements__(role="pointOfContact")
+
+    def __set_poc__(self, user_profile):
+        return self.__set_contact_role_element__(user_profile=user_profile, role="pointOfContact")
+
+    poc = property(__get_poc__, __set_poc__)
+
+    @property
+    def poc_csv(self):
```

Comment:
I'll evaluate changing the name on `poc_as_csv` for all the new configurations added to make it more explicit